README.md

```diff
@@ -71,14 +71,49 @@ int main(void)
 
 Refer to the [Quick Start Guide](http://github.com/thrust/thrust/wiki/Quick-Start-Guide) page for further information and examples.
 
-CMake Support
--------------
+Adding Thrust To A CMake Project
+--------------------------------
 
-Thrust provides CMake configuration files that make it easy to include Thrust
+Since Thrust is a header library, there is no need to build or install Thrust
+to use it. The `thrust` directory contains a complete, ready-to-use Thrust
+package upon checkout.
+
+We provide CMake configuration files that make it easy to include Thrust
 from other CMake projects. See the [CMake README](thrust/cmake/README.md)
 for details.
 
-Development process
+Development Process
 -------------------
 
-For information on development process, see [this document](doc/development_model.md).
+Thrust uses the [CMake build system](https://cmake.org/) to build unit tests,
+examples, and header tests. To build Thrust as a developer, the following
+recipe should be followed:
+
+```
+# Clone Thrust and CUB repos recursively:
+git clone --recursive https://github.com/thrust/thrust.git
+cd thrust
+
+# Create build directory:
+mkdir build
+cd build
+
+# Configure -- use one of the following:
+cmake ..   # Command line interface.
+ccmake ..  # ncurses GUI (Linux only)
+cmake-gui  # Graphical UI, set source/build directories in the app
+
+# Build:
+cmake --build -j .   # invokes make (or ninja, or msbuild, etc)
```

Comment:
I see this error message when I run this command:

```
jhoberock@jhoberock-dt:~/dev/git/thrust-jaredhoberock/build$ cmake --build -j .
Unknown argument .
Usage: cmake --build <dir> [options] [-- [native-options]]
Options:
  <dir>          = Project binary directory to be built.
  --parallel [<jobs>], -j [<jobs>]
                 = Build in parallel using the given number of jobs. 
                   If <jobs> is omitted the native build tool's 
                   default number is used.
                   The CMAKE_BUILD_PARALLEL_LEVEL environment variable
                   specifies a default parallel level when this option
                   is not given.
  --target <tgt>..., -t <tgt>... 
                 = Build <tgt> instead of default targets.
  --config <cfg> = For multi-configuration tools, choose <cfg>.
  --clean-first  = Build target 'clean' first, then build.
                   (To clean only, use --target 'clean'.)
  --verbose, -v  = Enable verbose output - if supported - including
                   the build commands to be executed. 
  --             = Pass remaining options to the native tool.
```

Reply:
Something like this worked, though:

```
cmake --build . -j 32
```

Reply:
Ah, right, the `.` is an arg to `--build`, not `cmake`, and the raw `-j` is ninja specific.

I should also mention that you can just call make / ninja / etc directly, depending on the generator used. I think that's a more common pattern.

I'll update these when I get back to this.